corehq/apps/app_manager/models.py

```diff
@@ -5961,21 +5961,11 @@ class LatestEnabledAppRelease(models.Model):
     activated_on = models.DateTimeField(null=True, blank=True)
     deactivated_on = models.DateTimeField(null=True, blank=True)
 
-    @staticmethod
-    def expire_cache(domain, location_id, app_id):
-        """
-        expire cache for all location and its descendants for the app
-        why? : Latest enabled release for a location is dependent on restrictions added for
-        itself and its ancestors. Hence we expire the cache for location and its descendants for which the
-        latest enabled release would depend on this location
-        :param domain: domain name
-        :param location_id: location for which the restriction is being updated
-        :param app_id: app id
-        """
-        location = SQLLocation.active_objects.get(location_id=location_id)
-        location_and_descendats = location.get_descendants(include_self=True)
-        for loc in location_and_descendats:
-            get_latest_enabled_app_release.clear(domain, loc.location_id, app_id)
+    def save(self, *args, **kwargs):
+        super(LatestEnabledAppRelease, self).save(*args, **kwargs)
+        from corehq.apps.app_manager import signals
+        signals.latest_enabled_app_release_post_save.send(LatestEnabledAppRelease,
```

Comment:
using a signal here is a bit confusing. I think signals are normally meant to be used to decouple things between apps that shouldn't really know anything about each other, but might be interested in the same events.
I would just call `expire_get_latest_enabled_app_release_cache` from here.

Reply:
[commit](https://github.com/dimagi/commcare-hq/pull/23680/commits/6bd9823e890b916e29347c1fd6ba9d86a8b5fe11)